src/http.ffi.mjs

```diff
@@ -88,17 +73,51 @@ const server = Http.createServer((req, res) => {
   }
 });
 
-export const serve = (host, port, on_start, on_port_taken) => {
-  let tries = 1;
-  server.on("error", (error) => {
-    if (error.code === "EADDRINUSE") {
-      let is_first_try = tries === 1;
-      if (is_first_try) {
-        on_port_taken(port);
+export const serve = (
+  { host, port, include_styles },
+  on_start,
+  on_port_taken
+) => {
+  let is_first_try = true;
+
+  html = `<!DOCTYPE html>
```

Comment:
I'm not really happy with how this works at all, the code is so ugly like this.

Reply:
I see, still I don't think I have any idea to make it less ugly 😬
Maybe for now a separate function where you can feed in extra things to cram in `<head>` could do it?